lib/trento.pm

```diff
@@ -826,6 +827,52 @@ sub cluster_wait_status {
     die "Timeout waiting for the change" if !$done;
 }
 
+=head3 cluster_wait_status_by_regex
+
+Remotly run 'SAPHanaSR-showAttr' in a loop on $host, wait output that match using regular expression
+=over 3
+
+=item B<HOST> - Ansible name or filter for the remote host where to run 'SAPHanaSR-showAttr'
+
+=item B<TIMEOUT> - Max time to retry. Die if timeout
+
+=item B<REGULAR_EXPRESSION> - Regular expression to match the text to find
+
+=back
+=cut
+
+sub cluster_wait_status_by_regex {
+    my ($host, $timeout, $regular_expression) = @_;
+    $timeout //= bmwqemu::scale_timeout(300);
+    my $_monitor_start_time = time();
+    my $prov = get_required_var('PUBLIC_CLOUD_PROVIDER');
+    my $done;
+    croak "Not regular expression provided" if ($regular_expression eq "") ;
+    while ((time() - $_monitor_start_time <= $timeout) && (!$done)) {
```

Comment:
`$done` used here but not initialized

Reply:
> `$done` used here but not initialized

No need to. It's defined in L849 without initialization, so it's `undef` which is `false` and `!$done` would be `true`.

Reply:
thx for the comment